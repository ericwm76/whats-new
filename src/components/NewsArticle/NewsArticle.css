.news-article {
  background-color: #9BDCFF;
  border: 1px solid black;
  border-radius: 5px;
  display: inline-block;
}

img {
  height: 100px;
  width: 100px;
}