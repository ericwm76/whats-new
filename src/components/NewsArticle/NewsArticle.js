import React from 'react';
import './NewsArticle.css';

const NewsArticle = (props) => {
  
  return (
    <article>
      <h3>{props.headline}</h3>
      <img src={props.img} alt=""/>
      <p>{props.description}</p>
      <a href={props.url} target="_blank" rel="noopener noreferrer">Link to Article</a>
      <button onClick="">X</button>
    </article>
  )
}

export default NewsArticle;