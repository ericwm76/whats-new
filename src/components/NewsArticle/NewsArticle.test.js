import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () =>{
  let mockArticle = { headline: "Harry Potter", description: "Wizarding is cool", img: "hogwarts", url: "https://www.wizardingworld.com/"};
  let wrapper = shallow(<NewsArticle headline={mockArticle.headline} img={mockArticle.img} description={mockArticle.description} url={mockArticle.url} key='1' id='1' />)
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})