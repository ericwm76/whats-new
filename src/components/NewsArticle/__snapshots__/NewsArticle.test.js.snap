// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot 1`] = `
<article
  className="news-article"
>
  <h3>
    Harry Potter
  </h3>
  <img
    alt=""
    src="hogwarts"
  />
  <p>
    Wizarding is cool
  </p>
  <a
    href="https://www.wizardingworld.com/"
    rel="noopener noreferrer"
    target="_blank"
  >
    Link to Article
  </a>
</article>
`;
